vendor/github.com/docker/go-connections/tlsconfig/config.go

```diff
@@ -83,6 +88,65 @@ func certPool(caFile string) (*x509.CertPool, error) {
 	return certPool, nil
 }
 
+func IsErrEncryptedKey(err error) bool {
+	return err != nil && strings.Contains(err.Error(), encryptedPrivateKeyMessage)
+}
```

Comment:
Once you switch to `errors.Wrap`, `getPrivateKey` can return a global error variable (`var errCantDecrypt = errors.New("private key is encrypted, but could not decrypt it")`), and then this function can compare `errors.Cause(err) == errCantDecrypt` instead of looking inside the string.